include/sqlpp11/sqlite3/connection.h

```diff
@@ -589,12 +512,110 @@ namespace sqlpp
 
         return {name};
       }
+
+    protected:
+      _handle_ptr_t _handle;
+
+      // Constructors
+      connection_base() = default;
+      connection_base(_handle_ptr_t&& handle) : _handle{std::move(handle)}
+      {
+      }
+
+    private:
+      enum class transaction_status_type
```

Comment:
Is there a technical reason for moving this code from above?

If no, then I would prefer to move the code in a separate PR.

Reply:
No real technical reason, so I moved the private part of the class back to the top. I also ran the SQLite3 connector tests and committed the update.

My rationale for moving the private part to the bottom was as follows:

- At the top we have some public types, like _config_t, _config_ptr_t, _handle_t, etc. Not all of them should be public but I left them public for convenience (mostly for simpler testing code).
- After we have defined the public types, we can use them to define the handle member `_handle_ptr_t _handle`. The handle object needs to use the public type _handle_ptr_t and it also should stay protected (no need to be public, but it should be accessible from the derived class pooled_connection, so the only choice is protected).
- After we have defined the _handle member we can use it in our private code. Technically our private code can use the _handle member even before it is defined, but arguably it the code is more readable if the member data is defined before the code that uses it.

So that is why I ended with following order 
public: -> protected: -> private:

Overall it is just a matter of preferences, so  I moved the private code back to the top to simplify the review of the code.

Reply:
Thanks! Can you do the same for the mysql connector as well (sorry for not explicitly asking for that before)?
This is a massive change. Reducing the size is important.

I am happy to discuss moving the code as you suggested in a second step :-)

Reply:
OK, I reordered back the code in the MySQL and PostgreSQL connectors too and also reverted a few code cleanups in the PostgreSQL connector in order to reduce the size of the diff.

Reply:
Great, thanks!